File: day1/day1.pl
print ("Advent of code: Day 1 - Perl\n");

open FILE, "input.txt" or die $!;

$floor = 0;
$basementFound = 0;
$basementPosition = 0;

my ($pos, $data, $n);
$pos = 0;
while (($n = read FILE, $data, 1) != 0) {
  if($data eq '(') {
    $floor++;
  } elsif($data eq ')') {
    $floor--;
  }
  if($floor == -1 && $basementFound == 0) {
    $basementFound = 1;
    $basementPosition = $pos+1;
  }
  $pos++;
}

print ("Part One: Santa ended up on floor: ", $floor, "\n");
print ("Part Two: Santa entered the basement on position ", $basementPosition, "\n");

close(FILE);

File: 2015/day1/day1.py
print "Advent of code: Day 1 - Python"

file = open('input.txt')
input = file.read()

ilen = len(input)

floor = 0
foundBasement = False
basementPosition = 0

for pos in range(ilen):
  move = input[pos]
  if move == '(':
    floor = floor+1
  if move == ')':
    floor = floor-1
  if floor == -1 and foundBasement is False:
      foundBasement = True
      basementPosition = pos+1

print "Part One: Santa ended up on floor: %r " % floor
print "Part Two: Santa entered the basement on position %r" % basementPosition
